### ISSUE TITLE
fallback in trojan not working

### ISSUE THREAD

**Issue by tarvfx** [SagerNet/sing-box#490]
### Welcome

- [X] Yes, I'm using the latest major release. Only such installations are supported.
- [X] Yes, I'm using the latest Golang release. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, **FULL** config, **FULL** log, etc).

### Description of the problem

Trojan can access the Internet, but fallback does't work. 

sing-box console output `connection closed: process connection from 127.0.0.1:39160: bad path: * | fallback connection: upload: EOF | download: EOF`

```log
Mar 31 17:58:08 sing-box[3004217]: INFO[2517] inbound/trojan[trojan-in]: fallback connection to mirror.coganng.com:443
Mar 31 17:58:08 sing-box[3004217]: INFO[2517] outbound/direct: outbound connection to mirror.coganng.com:443
Mar 31 17:58:08 sing-box[3004217]: DEBUG[2517] dns: lookup domain mirror.coganng.com
Mar 31 17:58:08 sing-box[3004217]: DEBUG[2517] dns: lookup succeed for mirror.coganng.com: 116.15.69.197
Mar 31 17:58:09 sing-box[3004217]: DEBUG[2518] inbound/trojan[trojan-in]: connection closed: process connection from 127.0.0.1:39160: bad path: * | fallback connection: upload: EOF | download: EOF
```

### Version of sing-box

```console
$ sing-box version
1.2.1
```

### Server and client configuration file

server:
```json
        {
            "type": "trojan",
            "tag": "trojan-in",
            "listen": "127.0.0.1",
            "listen_port": 5443,
            "tcp_fast_open": true,
            "udp_fragment": true,
            "proxy_protocol": true,
            "proxy_protocol_accept_no_header": true,
            "users": [
                {
                    "name": "my name",
                    "password": "my password"
                }
            ],
            "tls": {
                "enabled": true,
                "server_name": "my domain",
                "alpn": [
                    "h2",
                    "http/1.1"
                ],
                "min_version": "1.2",
                "max_version": "1.3",
                "certificate_path": "/usr/local/etc/sing-box/ssl/cert.pem",
                "key_path": "/usr/local/etc/sing-box/ssl/key.pem"
            },
            "fallback": {
                "server": "mirror.coganng.com",
                "server_port": 443
            },
            "transport": {
                "type": "ws",
                "path": "/bfe"
            }
        }
```

client:
```json
        {
            "type": "trojan",
            "tag": "trojan-out",
            "server": "my domain",
            "server_port": 443,
            "password": "my password",
            "tls": {
                "enabled": true
            },
            "multiplex": {
                "enabled": true,
                "protocol": "smux",
                "max_connections": 4,
                "min_streams": 4,
                "max_streams": 0
            },
            "transport": {
                "type": "ws",
                "path": "/bfe"
            }
        }
```

### Server and client log file

server console
```console
Mar 31 17:58:08 sing-box[3004217]: INFO[2517] inbound/trojan[trojan-in]: fallback connection to mirror.coganng.com:443
Mar 31 17:58:08 sing-box[3004217]: INFO[2517] outbound/direct: outbound connection to mirror.coganng.com:443
Mar 31 17:58:08 sing-box[3004217]: DEBUG[2517] dns: lookup domain mirror.coganng.com
Mar 31 17:58:08 sing-box[3004217]: DEBUG[2517] dns: lookup succeed for mirror.coganng.com: 116.15.69.197
Mar 31 17:58:09 sing-box[3004217]: DEBUG[2518] inbound/trojan[trojan-in]: connection closed: process connection from 127.0.0.1:39160: bad path: * | fallback connection: upload: EOF | download: EOF
```

**Comment by nekohasekai** [comment]
Fallback requests should connect to http servers, not https.

**Comment by tarvfx** [comment]
Hi, @nekohasekai 

After modifying the local http site, the problem still persists

> Fallback requests should send to HTTP servers, not HTTPS.

```json
            "fallback": {
                "server": "127.0.0.1",
                "server_port": 11080
            },

```

```log
Apr 05 14:54:54 sing-box[2002447]: INFO[0010] inbound/trojan[trojan-in]: fallback connection to 127.0.0.1:11080
Apr 05 14:54:54 sing-box[2002447]: INFO[0010] outbound/direct: outbound connection to 127.0.0.1:11080
Apr 05 14:54:54 sing-box[2002447]: DEBUG[0010] inbound/trojan[trojan-in]: connection closed: process connection from 127.0.0.1:37092: bad path: * | fallback connection: upload: EOF | download: EOF
```

**Comment by tarvfx** [comment]
![image](https://user-images.githubusercontent.com/70388114/230008320-e0776aa6-3a90-48e5-8b57-1b50ff631a47.png)
